streams/src/test/java/org/apache/kafka/streams/state/internals/GlobalStateStoreProviderTest.java

```diff
@@ -105,28 +105,19 @@ public void before() {
                 Serdes.String(),
                 Serdes.String()).build());
 
-        final ProcessorContextImpl mockContext = niceMock(ProcessorContextImpl.class);
-        expect(mockContext.applicationId()).andStubReturn("appId");
-        expect(mockContext.metrics())
-            .andStubReturn(
+        final ProcessorContextImpl mockContext = mock(ProcessorContextImpl.class);
+        when(mockContext.applicationId()).thenReturn("appId");
+        when(mockContext.metrics())
+            .thenReturn(
                 new StreamsMetricsImpl(new Metrics(), "threadName", StreamsConfig.METRICS_LATEST, new MockTime())
             );
-        expect(mockContext.taskId()).andStubReturn(new TaskId(0, 0));
-        expect(mockContext.recordCollector()).andStubReturn(null);
```

Comment:
Why aren't we stubbing this method to return null after moving to powermock?

Reply:
As far as I know the default return value of mocks is null when the return an object. For validation's sake, however, you are right. You either need to when this or verify this.

Reply:
I haven't stubbed `mockContext.recordCollector()` method because it's never actually called. `andStubReturn` method of EasyMock doesn't verify that the method is actually called it only provides a default return value without enforcing strict behavior. Replacing `andStubReturn` with `andReturn` makes tests fail as shown below. (To verify this you have to convert `mockContext` variable to class field and call `verify(mockContext) after each test).

![image](https://github.com/apache/kafka/assets/81428651/92e3cbe0-370f-4937-86a1-5e4353ec5f73)